help/md/srGeweke.md

```diff
@@ -1,9 +1,58 @@
 ## name
 srGeweke
 ## title
+Geweke stopping rule
+
 ## description
+Allow an MCMC run to terminate once the specified criterion has been met.
+
+Geweke (1992) proposed a convergence diagnostic for Markov chains based on a test for equality of the means of the first and last part of a Markov chain (by default the first 10% and the last 50%). If the samples are drawn from the stationary distribution of the chain, the two means are equal and Geweke's statistic has an asymptotically standard normal distribution.
+
+The test statistic is a standard Z-score: the difference between the two sample means divided by its estimated standard error. The standard error is estimated from the spectral density at zero and so takes into account any autocorrelation.
+
+The Z-score is calculated under the assumption that the two parts of the chain are asymptotically independent, which requires that the sum of `frac1` and `frac2` be strictly less than 1.
+
 ## details
+When passed as an element of the `StoppingRules[]` argument to `mcmc.run()` or `mcmcmc.run()`, the run will terminate once *all* specified convergence rules ([`srGelmanRubin()`], [`srGeweke()`], [`srMinESS()`], [`srStationarity()`]) have been fulfilled; or once *any* threshold rules ([`srMaxTime()`], [`srMaxIteration()`]) are met.
+
 ## authors
+Incorporates text by Martyn Plummer
+
 ## see_also
+
+- Tutorial on [convergence assessment](https://revbayes.github.io/tutorials/convergence/)
+
 ## example
+```
+# Specify a model
+mymodel = model(tree)
+
+# Define monitors to monitor parameters for convergence
+paramFile = "parameters.log"
+# Configure the parameter file to monitor only the parameters whose convergence shall be assessed
+monitors[1] = mnModel( filename = paramFile, printgen = 10, stochasticOnly = TRUE, exclude = ["rel_br_lengths"] )
+
+# Define threshold rules (optional): Stop as soon as ANY has been met
+stopping_rules[1] = srMaxIteration(200000)
+stopping_rules[2] = srMaxTime(15, "hours")
+
+# Check for convergence every 1000 iterations
+convergenceFreq = 1000
+
+# Add convergence rules: Stop as soon as ALL have been met
+stopping_rules[3] = srMinESS(50, file = paramFile, freq = convergenceFreq)
+stopping_rules[4] = srGelmanRubin(1.01, file = paramFile, freq = convergenceFreq)
+stopping_rules[5] = srGeweke(prob = 0.001, file = paramFile, freq = convergenceFreq)
+stopping_rules[6] = srStationarity(prob = 0.01, file= paramFile ,freq = convergenceFreq)
+
+mymcmc = mcmc(mymodel, monitors, moves)
+# Begin the MCMC run.
+# The checkpoint file will allow the run to be resumed from where it left off
+# if you later opt for more stringent stopping rules
+mymcmc.run(rules = stopping_rules, checkpointFile = "checkpoint.ckp", checkpointInterval = 1000)
```

Comment:
I think these examples should ideally be (1) self-contained, (2) simpler, and (3) specific to a given rule. What's given here is actually the most complex case possible, since all available stopping rules are used together. I'd suggest something like this:

```
# Binomial example: 7 successes out of 20 trials
r ~ dnExp(10)
p := Probability(ifelse(r < 1, r, 1))
n <- 20
k ~ dnBinomial(n, p)
k.clamp(7)

mymodel = model(k)

moves = VectorMoves()
moves.append( mvSlide(r, delta=0.1, weight=1) )

paramFile = "parameters.log"

monitors = VectorMonitors()
monitors.append( mnScreen(printgen=100, p) )
monitors.append( mnModel(filename=paramFile, printgen=100, p) )

# Stop when the test statistic becomes significant at alpha = 0.001
stopping_rules[1] = srGeweke(prob=0.001, file=paramFile, freq=10000)

# Create the MCMC object
mymcmc = mcmc(mymodel, monitors, moves)

# Begin the MCMC run
mymcmc.run(stopping_rules)
```

(This particular example apparently does well on `srGeweke` and `srMinESS` but terribly on `srGelmanRubin` and `srStationarity`. Even so, it's still useful for documentation purposes.)

Reply:
Yes, on reflection, I see that this is a better approach.  

In adapting this code for srMaxIteration, I wanted to add an explanation of why one might use srMaxIteration rather than just passing an `nGen` argument to `mcmc`.  I realise I'm not sure how the behaviour differs.  What is the motivation for a separate srMaxIteration function – do stopping rules cause `nGen` to be ignored?

Reply:
Hmm, good question. It looks like the `generations` (not `nGen`) argument behaves as just another stopping rule. If you combine it with `srMaxIteration` and give them different values, the total number of iterations will be determined by whichever is reached first. I guess the only point of `srMaxIteration` is that you can combine it with other stopping rules in the same vector and pass it to `mcmc.run()` as a single argument, instead of having to specify two arguments.